src/common/hooks/useGeolocation.ts

```diff
@@ -30,10 +37,16 @@ export default function useGeolocation(): GeolocationInfo | undefined {
             country: data.country,
           }),
         )
-        .catch(e => console.error(e));
+        .catch(e => console.error(e))
+        .finally(() => setIsLoading(false));
     };
     fetchIpData();
   }, []);
```

Comment:
`useEffect` need to have `setIpData` and `setIsLoading` as dependencies, also, we probably don't want to fetch the IP data if we have it already. If we don't add the dependencies the code might run more often that cause issues that are hard to debug

```ts
useEffect(() => {
  const fetchIpData = /* ... */;
  if (ipData) {
    fetchIpData();
  }
}, [setIpData, ipData, setIsLoading]);
```